src/Symfony/Component/RateLimiter/TokenBucket.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\RateLimiter;
+
+/**
+ * @author Wouter de Jong <wouter@wouterj.nl>
+ *
+ * @internal
```

Comment:
Why is it internal?

Reply:
If we're going to keep the factory, the token bucket is only used in `LimiterFactory` and `Limiter` - not in any userland code.